code/modules/mob/living/silicon/robot/drone/drone_console.dm

```diff
@@ -6,122 +6,132 @@
 	req_access = list(ACCESS_ENGINE_EQUIP)
 	circuit = /obj/item/circuitboard/drone_control
 
-	//Used when pinging drones.
-	var/drone_call_area = "Engineering"
-	//Used to enable or disable drone fabrication.
+	/// The linked fabricator
 	var/obj/machinery/drone_fabricator/dronefab
+	/// Used when pinging drones
+	var/drone_call_area = "Engineering"
+	/// Cooldown for area pings
+	var/ping_cooldown = 0
 
-/obj/machinery/computer/drone_control/attack_ai(var/mob/user as mob)
-	return src.attack_hand(user)
-
+/obj/machinery/computer/drone_control/Initialize(mapload)
+	. = ..()
+	find_fab()
 
-/obj/machinery/computer/drone_control/attack_hand(var/mob/user as mob)
+/obj/machinery/computer/drone_control/attack_hand(mob/user)
 	if(..())
 		return
 
 	if(!allowed(user))
 		to_chat(user, "<span class='warning'>Access denied.</span>")
 		return
 
-	interact(user)
-
-/obj/machinery/computer/drone_control/attack_ghost(mob/user as mob)
-	interact(user)
-
-/obj/machinery/computer/drone_control/interact(mob/user)
-
-	user.set_machine(src)
-	var/dat
-	dat += "<B>Maintenance Units</B><BR>"
-
+	ui_interact(user)
+
+/obj/machinery/computer/drone_control/attack_ghost(mob/user)
+	ui_interact(user)
+
+// tgui\packages\tgui\interfaces\DroneConsole.js
+/obj/machinery/computer/drone_control/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = TRUE, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state)
```

Comment:
I would set `force_open` to `FALSE` to prevent recreating the window every time someone interacts with the machine, instead refreshing the existing window.